### DISCUSSION TITLE
Tenant is not allowing authentication

### DISCUSSION THREAD

**Discussion by LesSpangler** [svrooij/WinTuner#56] [Q&A]
Publish works perfectly up to authentication.
I then get a message indicating:
User "Needs admin approval"
unverified
needs permission to access resources in your organization that only an admin can grant.

Ideas where in the tenant this would be set..      The username I am using has full "Intune Administrator" rights in the tenant.

**Comment by svrooij** [comment]
You global administrator can [configure](https://learn.microsoft.com/en-us/entra/identity/enterprise-apps/user-admin-consent-overview?wt.mc_id=SEC-MVP-5004985) who can grant access to new applications. There is also an option that users (like you) can request access to an app.

## Admin consent for 1 user

Microsoft published a document on how to grant access to an application for a single user, check [this documentation page](https://learn.microsoft.com/en-us/entra/identity/enterprise-apps/grant-consent-single-user?pivots=msgraph-powershell&wt.mc_id=SEC-MVP-5004985)

But it needs some changes:

- `$clientAppId` must be `d5a8a406-3b1d-4069-91cc-d76acdd812fe`
- `$userUpnOrId` must be your account
- `$permissions` needs to have these added `,"DeviceManagementConfiguration.ReadWrite.All", "DeviceManagementApps.ReadWrite.All"` 

The script will create a service principal if required and will assign the app to you.

## Admin consent:

1. Global admin should consent to [WinTuner by @svrooij consent link](https://login.microsoftonline.com/organizations/adminconsent?client_id=d5a8a406-3b1d-4069-91cc-d76acdd812fe)
2. (optional) Go to Enterprise applications and switch this application to requires User assignment
3. Under Enterprise applications assign the application to your account

## Own app registration

Or if you don't want to use my app registration for some reason, you can ask someone with the right access to create a new app registration and assign that. Then you just have to add the client id to each command.

When configuring the new application be sure to set in as native application with these redirect urls:
![image](https://github.com/svrooij/WingetIntune/assets/1292510/3ba9a577-41c4-4b0c-88c2-a8296c5db2d8)
